src/crypto/crypto_keys.cc

```diff
@@ -815,6 +838,51 @@ void KeyObjectHandle::InitEDRaw(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(true);
 }
 
+#if OPENSSL_VERSION_MAJOR >= 3 && OPENSSL_VERSION_MINOR >= 5
+void KeyObjectHandle::InitMlDsaRaw(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  KeyObjectHandle* key;
+  ASSIGN_OR_RETURN_UNWRAP(&key, args.This());
+
+  CHECK(args[0]->IsString());
+  Utf8Value name(env->isolate(), args[0]);
+
+  ArrayBufferOrViewContents<unsigned char> key_data(args[1]);
+  KeyType type = FromV8Value<KeyType>(args[2]);
+
+  MarkPopErrorOnReturn mark_pop_error_on_return;
+
+  typedef EVPKeyPointer (*new_key_fn)(
+      int, const ncrypto::Buffer<const unsigned char>&);
+  new_key_fn fn = type == kKeyTypePrivate ? EVPKeyPointer::NewRawSeed
+                                          : EVPKeyPointer::NewRawPublic;
+
+  int id = GetNidFromName(*name);
+
+  switch (id) {
+    case EVP_PKEY_ML_DSA_44:
+    case EVP_PKEY_ML_DSA_65:
+    case EVP_PKEY_ML_DSA_87: {
+      auto pkey = fn(id,
+                     ncrypto::Buffer<const unsigned char>{
+                         .data = key_data.data(),
+                         .len = key_data.size(),
+                     });
+      if (!pkey) {
+        return args.GetReturnValue().Set(false);
+      }
+      key->data_ = KeyObjectData::CreateAsymmetric(type, std::move(pkey));
+      CHECK(key->data_);
+      break;
+    }
+    default:
+      UNREACHABLE();
```

Comment:
Just for my own understanding, is this different than the `unreachable();` used previously? What is the difference (if any)?

Reply:
Okay cool. It really is just different casing so I wasn't sure. Maybe i'll dig into later and try to figure it out.